static/js/muting_ui.js

```diff
@@ -187,11 +195,41 @@ export function handle_user_updates(muted_user_ids) {
         return;
     }
     muting.set_muted_users(muted_user_ids);
+    rerender_on_user_update();
+}
+
+export function set_up_muted_users_ui() {
+    const muted_users = muting.get_muted_users().map((user) => ({
+        user_id: user.id,
+        user_name: people.get_full_name(user.id),
+        date_muted_str: user.date_muted_str,
+    }));
+    const muted_users_table = $("#muted_users_table");
+    const $search_input = $("#muted_users_search");
+
+    ListWidget.create(muted_users_table, muted_users, {
+        name: "muted-users-list",
+        modifier(muted_users) {
+            return render_muted_user_ui_row({muted_users});
+        },
+        filter: {
+            element: $search_input,
+            predicate(item, value) {
+                return item.user_name.toLocaleLowerCase().includes(value);
+            },
+            onupdate() {
+                ui.reset_scrollbar(muted_users_table.closest(".progressive-table-wrapper"));
+            },
+        },
+        parent_container: $("#muted-user-settings"),
+        simplebar_container: $("#muted-user-settings .progressive-table-wrapper"),
+    });
 }
```

Comment:
This function more naturally belongs in `settings_muted_users.js`; can you move it?

And maybe add a prep commit to move the similar function to `settings_muted_topics.js`.

Reply:
I think `rerender_on_user_update` has the same story.

Reply:
> And maybe add a prep commit to move the similar function to settings_muted_topics.js.

#18279 handles this.

> I think rerender_on_user_update has the same story.

Not really, that function handles a lot more stuff, not just the settings page.